# - Amazon Skincare Products - 
# Analyse exploratoire des données 

### Ce dataset présente des produits cosmétiques issus du site Amazon et comprend des détails sur chaque produit : titre, type de produit, catégorie de produits, marque, type de peau, nombre d'items, nom du commerçant, son prix et le lien vers sa fiche produit.

### L'analyse de ce dataset s'inscrit dans le cadre du projet d'entraînement d'un modèle permettant de proposer des produits adaptés à des problématiques de peau après soumission d'une photo selfie par l'utilisateur.

## Import des données

In [11]:
import pandas as pd
df = pd.read_csv(r"data\raw\skincare_products\skincare.csv")
df

FileNotFoundError: [Errno 2] No such file or directory: 'data\\raw\\skincare_products\\skincare.csv'

## Analyse générale du dataset

In [ ]:
print("Nombre de lignes : ", df.shape[0])
print("Nombre de colonnes : ", df.shape[1])

Nombre de lignes :  1438
Nombre de colonnes :  10


In [ ]:
# aperçu des 5 premières lignes
df.head()

,Unnamed: 0,Title,Product,Category,Brand,Skin_Type,Number of items,Sold By,Price,Link
0,0,"Nivea Soft Light Moisturizer For Face,...",Moisturizer,SKINCARE,Nivea,All,1,AIR SALES,372,https://www.amazon.in/sspa/click?ie=UTF8&spc=M...
1,1,"Nivea Soft Playful Peach, Light Moistu...",Moisturizer,SKINCARE,Nivea,All,1,RK World Infocom Pvt Ltd,226,https://www.amazon.in/sspa/click?ie=UTF8&spc=M...
2,2,"Nivea Soft Light Moisturizer For Face,...",Moisturizer,SKINCARE,Nivea,Dry,1,MURALI RETAILS,103,https://www.amazon.in/sspa/click?ie=UTF8&spc=M...
3,3,Biotique Morning Nectar Flawless Skin ...,Moisturizer,SKINCARE,Biotique,Combination,1,RK World Infocom Pvt Ltd,240,https://www.amazon.in/Biotique-Morning-Nectar-...
4,4,Minimalist Marula Oil 5% Face Moisturi...,Moisturizer,SKINCARE,Minimalist,Dry,1,Minimalist Inc,284,https://www.amazon.in/Minimalist-Moisturizer-H...


In [ ]:
# types de données
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 1438 entries, 0 to 1437
Data columns (total 10 columns):
 #   Column           Non-Null Count  Dtype
---  ------           --------------  -----
 0   Unnamed: 0       1438 non-null   int64
 1   Title            1438 non-null   str  
 2   Product          1438 non-null   str  
 3   Category         1438 non-null   str  
 4   Brand            1438 non-null   str  
 5   Skin_Type        1438 non-null   str  
 6   Number of items  1438 non-null   int64
 7   Sold By          1231 non-null   str  
 8   Price            1438 non-null   int64
 9   Link             1438 non-null   str  
dtypes: int64(3), str(7)
memory usage: 112.5 KB


Les types de données sont des nombres entiers et des chaînes de caractères





In [ ]:
df.describe(include="all")

,Unnamed: 0,Title,Product,Category,Brand,Skin_Type,Number of items,Sold By,Price,Link
count,1438.000000,1438,1438,1438,1438,1438,1438.000000,1231,1438.000000,1438
unique,NaN,1268,5,1,509,57,NaN,403,NaN,1438
top,NaN,Kimirica Hand Creams Combo gifting | M...,Nail paint,SKINCARE,Generic,All,NaN,RK World Infocom Pvt Ltd,NaN,https://www.amazon.in/sspa/click?ie=UTF8&spc=M...
freq,NaN,4,299,1438,43,992,NaN,393,NaN,1
mean,718.500000,NaN,NaN,NaN,NaN,NaN,1.563978,NaN,619.605702,NaN
std,415.259156,NaN,NaN,NaN,NaN,NaN,13.273306,NaN,1481.257665,NaN
min,0.000000,NaN,NaN,NaN,NaN,NaN,1.000000,NaN,45.000000,NaN
25%,359.250000,NaN,NaN,NaN,NaN,NaN,1.000000,NaN,199.000000,NaN
50%,718.500000,NaN,NaN,NaN,NaN,NaN,1.000000,NaN,296.500000,NaN
75%,1077.750000,NaN,NaN,NaN,NaN,NaN,1.000000,NaN,476.000000,NaN


On distingue : 
- 1438 lignes
- 1268 titres uniques
- 509 marques uniques
- 57 types de peau

Le dataset est riche, ses catégories sont équilibrées, il y a une diversité des marques. Cependant la colonne Skin_Type possède problablement des données incohérentes et les prix ont des valeurs extrêmes.

In [ ]:
# nombre de valeurs manquantes
df.isnull().sum() 

Unnamed: 0           0
Title                0
Product              0
Category             0
Brand                0
Skin_Type            0
Number of items      0
Sold By            207
Price                0
Link                 0
dtype: int64

In [ ]:
# taux de valeurs manquantes dans le dataset
df.isnull().mean() * 100

Unnamed: 0          0.000000
Title               0.000000
Product             0.000000
Category            0.000000
Brand               0.000000
Skin_Type           0.000000
Number of items     0.000000
Sold By            14.394993
Price               0.000000
Link                0.000000
dtype: float64

Seule la colonne "Sold by" présente de données manquantes, à hauteur de 14,39%.

In [ ]:
# identification de valeurs cachées
(df == "").sum()

Unnamed: 0         0
Title              0
Product            0
Category           0
Brand              0
Skin_Type          0
Number of items    0
Sold By            0
Price              0
Link               0
dtype: int64

In [ ]:
# identification générale des doublons 
df.duplicated().sum()

np.int64(0)

In [ ]:
# identification des doublons par nom de produit (ciblé sur "Title")
df_duplicates_by_title = df[df.duplicated(subset=["Title"])]
df_duplicates_by_title

,Unnamed: 0,Title,Product,Category,Brand,Skin_Type,Number of items,Sold By,Price,Link
24,24,Neutriderm Moisturising Lotion w/Vitam...,Moisturizer,SKINCARE,Neutriderm,All,1,Iberia Skinbr,1000,https://www.amazon.in/sspa/click?ie=UTF8&spc=M...
32,32,"Nivea Soft Light Moisturizer For Face,...",Moisturizer,SKINCARE,Nivea,All,1,AIR SALES,372,https://www.amazon.in/Nivea-Light-Moisturising...
46,46,"Nivea Soft Light Moisturizer For Face,...",Moisturizer,SKINCARE,Nivea,Dry,1,MURALI RETAILS,103,https://www.amazon.in/Nivea-Soft-Cr%C3%A8me-50...
55,55,"Nivea Soft Playful Peach, Light Moistu...",Moisturizer,SKINCARE,Nivea,All,1,RK World Infocom Pvt Ltd,226,https://www.amazon.in/Nivea-Moisturizer-Playfu...
59,59,Neutrogena Hydro Boost Hyaluronic Acid...,Moisturizer,SKINCARE,Neutrogena,All,1,Global_India,786,https://www.amazon.in/sspa/click?ie=UTF8&spc=M...
...,...,...,...,...,...,...,...,...,...,...
1401,1401,JUICE Nail Polish Coral Sunset - 292 /...,Nail paint,SKINCARE,JUICE,All,1,JK Lifestyle Pvt Ltd.,199,https://www.amazon.in/sspa/click?ie=UTF8&spc=M...
1417,1417,GUBB Nail Paint Remover Dip & Twist | ...,Nail paint,SKINCARE,GUBB,All,1,GROOM TOWN PRIVATE LIMITED,179,https://www.amazon.in/GUBB-Acetone-Remover-Van...
1420,1420,MyGlamm LIT Nail Enamel-Say No More (R...,Nail paint,SKINCARE,MyGlamm,All,1,RK World Infocom Pvt Ltd,261,https://www.amazon.in/MyGlamm-Enamel-Say-Polis...
1435,1435,"JUICE Quick-dry, Nail Paint, Long Last...",Nail paint,SKINCARE,JUICE,All,1,JK Lifestyle Pvt Ltd.,199,https://www.amazon.in/sspa/click?ie=UTF8&spc=M...


In [ ]:
# Vérfication si les doublons identifiés sont des produits identiques vendus par différents commerçants
df_duplicates_soldby_excl = df[df.duplicated(subset=df.columns.difference(["Sold By"]), keep=False)]
df_duplicates_soldby_excl

,Unnamed: 0,Title,Product,Category,Brand,Skin_Type,Number of items,Sold By,Price,Link


Les doublons identifiés sur la colonne du nom de produit n'ont pas de lien avec des produits identiques vendus par des commerçants différents.
On peut conclure qu'il n'y a pas de véritable doublon mais des produits au nom proche.

## Analyse de la colonne Produit

In [ ]:
# répartition des produits par type
df["Product"].value_counts()

Product
Nail paint       299
Moisturizer      289
Primer           289
Facial sprays    282
Hand cream       279
Name: count, dtype: int64

## Analyse de la colonne Type de peau

In [ ]:
# liste des valeurs uniques et leur nombre
df["Skin_Type"].value_counts()

Skin_Type
All                                                    992
Normal                                                 102
Dry                                                     93
Combination                                             56
Sensitive                                               26
Oily                                                    22
All, Oily, Sensitive, Dry, Normal                       19
All, Oily, Combination, Sensitive, Dry                  14
All, Oily, Combination, Dry, Normal                     13
Oily, Combination, Dry, Normal                           9
All, Combination, Sensitive, Dry, Normal                 6
Oily, Combination, Sensitive, Dry, Normal                6
Sensitive, Dry                                           5
Acne Prone                                               5
Acne Prone Skin                                          5
All, Dry                                                 4
Acne Prone,All Skin Types                     

In [ ]:
# liste des valeurs uniques dans "Type de produit"
df["Skin_Type"].unique()

<StringArray>
[                                                'All',
                                                 'Dry',
                                         'Combination',
            'All, Combination, Sensitive, Dry, Normal',
                                              'Normal',
                                      'Sensitive, Dry',
                                    'Oily, Acne Prone',
                                         'Dry, Normal',
                                                'Acne',
                                                'Oily',
                         'Oily, Acne Prone, Sensitive',
                                           'All, Acne',
                                    'Dry, Normal Skin',
                                            'All, Dry',
                                           'Sensitive',
                      'Oily, Combination, Dry, Normal',
                                  'Normal to Dry skin',
           'All, Oily, Combination

# Nettoyage des données

### Suppression des colonnes inutiles

In [ ]:
df = df.drop(["Unnamed: 0", "Category", "Sold By", "Link", "Number of items"], axis=1)

In [ ]:
df

,Title,Product,Brand,Skin_Type
0,"Nivea Soft Light Moisturizer For Face,...",Moisturizer,Nivea,All
1,"Nivea Soft Playful Peach, Light Moistu...",Moisturizer,Nivea,All
2,"Nivea Soft Light Moisturizer For Face,...",Moisturizer,Nivea,Dry
3,Biotique Morning Nectar Flawless Skin ...,Moisturizer,Biotique,Combination
4,Minimalist Marula Oil 5% Face Moisturi...,Moisturizer,Minimalist,Dry
...,...,...,...,...
1433,PAVITYAKSH Saloon Express Stamping Pro...,Nail paint,PAVITYAKSH,All
1434,"JUICE Quick-dry, 100% Veg & Cruelty Fr...",Nail paint,JUICE,All
1435,"JUICE Quick-dry, Nail Paint, Long Last...",Nail paint,JUICE,All
1436,"BAD COMPANY No Toxin Nail Lacquer, Lon...",Nail paint,BAD COMPANY,All


Plusieurs colonnes ici sont supprimées car elles n'ont soit pas d'intérêt pour l'entraînement du modèle ("Unnamed: 0"), soit présentent des valeurs incohérentes (1 seule valeur pour "Category"), soit n'apparaîtront pas dans les informations des produits communiquées à l'étape des résultats ("Sold By", "Link", "Number of items").

### Suppression des lignes inutiles

In [ ]:
df = df[~df["Product"].isin(["Nail paint", "Hand cream"])]
df

,Title,Product,Brand,Skin_Type
0,"Nivea Soft Light Moisturizer For Face,...",Moisturizer,Nivea,All
1,"Nivea Soft Playful Peach, Light Moistu...",Moisturizer,Nivea,All
2,"Nivea Soft Light Moisturizer For Face,...",Moisturizer,Nivea,Dry
3,Biotique Morning Nectar Flawless Skin ...,Moisturizer,Biotique,Combination
4,Minimalist Marula Oil 5% Face Moisturi...,Moisturizer,Minimalist,Dry
...,...,...,...,...
855,evanhealy Heat of Summer HydroSoul Fac...,Facial sprays,Evanhealy,All
856,"D'MOKSHA Wine Vine Face Toner, Face Mi...",Facial sprays,D'MOKSHA,Acne Prone Skin
857,Love Earth Jasmine Mogra Face Mist Ton...,Facial sprays,Love Earth,All
858,EcstaCBeauty Cure Skin Anti-Breakout F...,Facial sprays,EcstaCBeauty,Sensitive


Les lignes dont les valeurs correspondent à "Nail paint" et "Hand cream" dans la colonne "Product" sont supprimées car elles ne font pas partie des produits proposés.

### Nombre d'éléments restants par type de produit

In [ ]:
df["Product"].value_counts()

Product
Moisturizer      289
Primer           289
Facial sprays    282
Name: count, dtype: int64

## Marques les plus représentées dans le dataset

In [ ]:
df["Brand"].value_counts()

Brand
Nivea         34
Generic       30
DEVIL MONK    22
LAKMÉ         18
Mamaearth     14
              ..
NOIANT         1
Evanhealy      1
D'MOKSHA       1
Love Earth     1
Isntree        1
Name: count, Length: 361, dtype: int64

Nivea, Devil Monk, Lakmé, Mamaearth et des marques génériques font partie des plus représentées.

### Nettoyage des valeurs de la colonne "Skin type"

In [ ]:
df["Skin_Type"] = (
    df["Skin_Type"]
    .str.lower()
    .str.strip()
    .str.replace(r"<.*?>", "", regex=True)
)

In [ ]:
df

,Title,Product,Brand,Skin_Type
0,"Nivea Soft Light Moisturizer For Face,...",Moisturizer,Nivea,all
1,"Nivea Soft Playful Peach, Light Moistu...",Moisturizer,Nivea,all
2,"Nivea Soft Light Moisturizer For Face,...",Moisturizer,Nivea,dry
3,Biotique Morning Nectar Flawless Skin ...,Moisturizer,Biotique,combination
4,Minimalist Marula Oil 5% Face Moisturi...,Moisturizer,Minimalist,dry
...,...,...,...,...
855,evanhealy Heat of Summer HydroSoul Fac...,Facial sprays,Evanhealy,all
856,"D'MOKSHA Wine Vine Face Toner, Face Mi...",Facial sprays,D'MOKSHA,acne prone skin
857,Love Earth Jasmine Mogra Face Mist Ton...,Facial sprays,Love Earth,all
858,EcstaCBeauty Cure Skin Anti-Breakout F...,Facial sprays,EcstaCBeauty,sensitive


Un travail de structuration sur la colonne "Skin_Type" est nécessaire car les données avaient une forme et un contenu initialement très variés d'une ligne à l'autre.

### En résumé, ce dataset est riche et exploitable, mais un nettoyage ciblé permettra d’en tirer tout le potentiel pour l’analyse exploratoire ou l’ingestion dans un pipeline RAG.

In [ ]:
df.to_csv("amazon_skincare_cleaned.csv", index=False)